server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1113,6 +1113,9 @@ public CanMatchResponse canMatch(ShardSearchRequest request) throws IOException
         assert request.searchType() == SearchType.QUERY_THEN_FETCH : "unexpected search type: " + request.searchType();
         IndexService indexService = indicesService.indexServiceSafe(request.shardId().getIndex());
         IndexShard indexShard = indexService.getShard(request.shardId().getId());
+        if (indexShard.isSearchIdle()) {
```

Comment:
Do we also need to check whether there are changes that have not been refreshed? Otherwise this change will have a terrible effiect on the warm tier by returning `can_match=true` up to once per second even though no changes are being made to indices? Or are we already protected against this?

Reply:
Good catch. I pushed https://github.com/elastic/elasticsearch/pull/55428/commits/0e7f9767583ccf66d89eb8600fa9c234b83cb8ac.

Reply:
I wonder if we can still extract the min/max values of the primary sort since we don't need the values to be accurate and use it only to order the shards execution. If indices are in the warm tier they should contain some recent data that we'd like to visit first if the query is sorted by timestamp.

Reply:
This sounds potentially dangerous to me. I wouldn't think of the special case of search-idle indices if someone raised an idea that would leverage min/max values and rely on the fact that the max value to be an upper bound and the min value to be a lower bound?

Reply:
We don't have this guarantee today either, the reader of the query phase can be different from the one used in the can match phase. I can update the comments to make it more clear in the code that we don't create a point in time reader during the can match phase.

Reply:
I agree that the `can_match` phase might get a different reader from the `query` phase, but I'm less concerned by this since the reader we get in the `can_match` phase would also be a legal reader to run the query phase on, which isn't the case here? I don't feel strongly about this, ok to return the min/max values and +1 to document this.

Reply:
Right, I also thought about this because having these guarantees would mean that we can filter shards on the coord node based on these values. We don't do this today because of search-idle shards so we delay the skipping when verifying the values locally during the query phase. 
The other option that was proposed by Nhat in this pr was to perform the refresh of search-idle shards during the can_match phase. It has the advantage of making all min/max values accurate for the search request but could make the phase much slower.

>  I don't feel strongly about this, ok to return the min/max values and +1 to document this.

++, thanks

Reply:
Thanks Adrien and Jim for the discussion here. I pushed https://github.com/elastic/elasticsearch/pull/55428/commits/80f492ff64e41d1f78a58a4cd537f7396a3230cc.